back-end/src/main/kotlin/Gneiss/PacketCompiler/Service/Users.kt

```diff
@@ -1,12 +1,36 @@
 package Gneiss.PacketCompiler.Service
 
-class CreateUserRequest()
-class CreateUserResponse()
+import Gneiss.PacketCompiler.DatabaseAccess.UserDao
+import org.springframework.http.*
 
-class Users {
-    companion object {
-        fun createUser(req: CreateUserRequest): CreateUserResponse {
-            return CreateUserResponse()
+class CreateUserRequest(
+        var email: String,
+        var password: String,
+        var firstName: String,
+        var lastName: String,
+        var roleId: Int,
+)
+
+class CreateUserResponse(
+        var responseString: String
+)
+
+class Users(userDao: UserDao) {
+    var userDao = userDao
+
+    fun createUser(req: CreateUserRequest): CreateUserResponse {
+        //Start by checking if there is already a user with the email specified in the CreateAccountRequest
+        //If there is create and return a HTTP error response
+        //Assume there is an account, and if not set the flag to false with checkAccountExists()
+        var accountExistsFlag: Boolean = true
+        accountExistsFlag = userDao.checkAccountExists(req.email)
+
+        if(accountExistsFlag == true) {
+            return CreateUserResponse("Account already exists for this email")
```

Comment:
This is still a 200 isn't it? If you throw an exception we can check by error code in the front end rather than having to compare strings, which would really couple our front and back ends.